crates/stc_ts_file_analyzer/src/analyzer/assign/mod.rs

```diff
@@ -2067,6 +2067,18 @@ impl Analyzer<'_, '_> {
                         // TODO: Handle Type::Rest
 
                         if elems.len() > rhs_elems.len() {
+                            let is_lhs_any_tuple = elems.iter().all(|l| match *l.ty {
```

Comment:
Is this correct? I think we should leverage the optional type as you said in https://github.com/dudykr/stc/issues/181#issuecomment-1306378999